File: P/1566_Detect_Pattern_of_Length_M_Repeated_K_or_More_Times.java
class Solution {
    public boolean containsPattern(int[] arr, int m, int k) {
        boolean found;
        for(int i = 0; i <= arr.length - m * k; i++) {
            found = true;
            for(int j = 1; found && j < k; j++) {
                for(int l = 0; found && l < m; l++) {
                    if(arr[i + l] != arr[i + l + (m * j)])
                        found = false;
                }
            }
            if(found)
                return found;
        }
        return false;
    }
}

File: 주영/Any/1566. Detect Pattern of Length M Repeated K or More Times.js
function containsPattern(arr, length, repetition) {
  const delta = length * repetition;

  for (let i = 0; i + delta <= arr.length; i++) {
    const pattern = arr.slice(i, i + delta).join('');

    if (pattern.slice(0, length).repeat(repetition) === pattern) {
      return true;
    }
  }

  return false;
}
